hazelcast/src/main/java/com/hazelcast/query/impl/IndexImpl.java

```diff
@@ -165,7 +166,7 @@ public boolean isOrdered() {
         return indexStore.getRecordMap(indexValue);
     }
 
-    public static final class NullObject implements Comparable, DataSerializable {
+    public static final class NullObject implements Comparable, IdentifiedDataSerializable {
```

Comment:
I don't know how frequently this object is being send, but perhaps we could make a cached version in the corresponding factory?